src/meta/service/src/ddl_service.rs

```diff
@@ -241,6 +242,7 @@ impl DdlService for DdlServiceImpl {
                         fragment_graph,
                         CreateType::Foreground,
                         None,
+                        HashSet::new(), // TODO(rc): pass dependencies through this field instead of `PbSource`
```

Comment:
👍 connection and secret are handled in separated fields now and can be unified into `dependencies`. This also applies to the connection for the sink.